src/components/controls/selects/MultiSelect/MultiSelectInput.tsx

```diff
@@ -0,0 +1,469 @@
+    onSearchChange?.(text)
+    onSearchChange(text)
+    onFocusToggle(true)
+    onIndexHighlight(null)
+  }
+
+  const handleInputFocus = (): void => {
+    onFocusToggle(true)
+  }
+
+  const handleInputClick = (): void => {
+    onIndexHighlight(null)
+  }
+
+  const handleInputKeyDown = (e: KeyboardEvent<HTMLInputElement>): void => {
+    const { current } = inputRef
+    if (
+      (e.key === "ArrowLeft" || e.key === "Backspace") &&
+      current?.selectionStart === 0 &&
+      current?.selectionEnd === 0
+    ) {
+      e.preventDefault()
+      if (highlightedIndex === null) {
+        onIndexHighlight(selectedOptions.length - 1)
+      } else if (highlightedIndex > 0) {
+        onIndexHighlight(highlightedIndex - 1)
+      }
+      setDirection("start")
+    }
+
+    if (e.key === "Backspace" && highlightedIndex !== null) {
+      const newOptions = selectedOptions.filter((_, index) => index !== highlightedIndex)
+      onSelectionChange(newOptions)
+      onIndexHighlight(null)
+    }
+
+    if (
+      e.key === "ArrowRight" &&
+      highlightedIndex !== null &&
+      highlightedIndex < selectedOptions.length - 1
+    ) {
+      e.preventDefault()
+      onIndexHighlight(highlightedIndex + 1)
+      setDirection("end")
+    } else if (e.key === "ArrowRight") {
+      if (highlightedIndex === selectedOptions.length - 1) {
+        e.preventDefault()
+      }
+      onIndexHighlight(null)
+    }
+  }
+
+  const handleClearInput = (
+    e: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>,
+  ): void => {
+    e.stopPropagation()
+    e.preventDefault()
+    if (!selectedOptions.length && !inputValue) {
+      inputRef.current?.blur()
+      onFocusToggle(false)
+      const button = e.currentTarget
+      button.blur()
+    } else {
+      inputRef.current?.focus()
+    }
+    onSelectionChange([])
+    onSearchChange("")
+    onSearchChange?.("")
+    onClearClick?.()
+  }
+
+  /**
+   * Focus input field when clicking on component
+   * @param e
+   */
+  const handlePillMouseDown = (e: MouseEvent<HTMLButtonElement>): void => {
+    e.preventDefault()
+  }
+
+  const handlePillClick = (index: number): void => {
+    if (isOpen) {
+      onIndexHighlight(index)
+    }
+  }
+
+  /**
+   * Don't blur input field on label click
+   * @param e
+   */
+  const handleLabelMouseDown = (e: MouseEvent<HTMLLabelElement>): void => {
+    if (isOpen) {
+      e.preventDefault()
+    }
+  }
+
+  return (
+    <>
+      {"label" in props && (
+        <StyledLabel {...props.labelProps} htmlFor={id} onMouseDown={handleLabelMouseDown}>
+          {props.label}
+        </StyledLabel>
+      )}
+      <Wrapper
+        role="combobox"
+        aria-haspopup
+        aria-expanded={isOpen}
+        aria-controls={`options-${id}`}
+        hasLabel={"label" in props}
+      >
+        <Scroller ref={scrollableRef}>
+          <ScrollContent style={{ flex: `0 0 ${contentInlineSize}px` }}>
```

Comment:
If would be nice to let `<ScrollContent>` extend `<Box>` here, so that you can make this instead:

```suggestion
          <ScrollContent flexBasis={`${contentInlineSize}px`}>
```

Reply:
This would create a lot of css classes while adding content to the input field:
https://emotion.sh/docs/best-practices#use-the-style-prop-for-dynamic-styles